.github/ISSUE_TEMPLATE/release-checklist.md

```diff
@@ -0,0 +1,50 @@
+---
+name: Zarr-Python release checklist
+about: Checklist for a new Zarr-Python release. [For project maintainers only!]
+title: Release Zarr-Python vX.Y.Z
+labels: release-checklist
+assignees: ''
+
+---
+
+**Release**: [v0.x.x](https://github.com/zarr-developers/zarr-python/milestones/?)
+**Scheduled Date**: 20YY/MM/DD
+
+**Priority PRs/issues to complete prior to release**
+
+- [ ] Priority pull request #X
+
+**Before release**:
+
+- [ ] Make sure that all pull requests which will be included in the release have been properly documented as changelog files in the [`changes/` directory](https://github.com/zarr-developers/zarr-python/tree/main/changes).
+- [ ] Run ``towncrier build --version x.y.z`` to create the changelog, and commit the result to the main branch.
+- [ ] Check [SPEC 0](https://scientific-python.org/specs/spec-0000/#support-window) to see if the minimum supported version of Python or NumPy needs bumping.
+- [ ] Check to ensure that:
+  - [ ] Deprecated workarounds/codes/tests are removed. Run `grep "# TODO" **/*.py` to find all potential TODOs.
+  - [ ] All tests pass in the ["Tests" workflow](https://github.com/zarr-developers/zarr-python/actions/workflows/test.yml)
+  - [ ] All tests pass in the ["GPU Tests" workflow](https://github.com/zarr-developers/zarr-python/actions/workflows/gpu_test.yml)
+  - [ ] All tests pass in the ["Hypothesis" workflow](https://github.com/zarr-developers/zarr-python/actions/workflows/hypothesis.yaml)
+  - [ ] All tests pass in [Xarray's upstream test](https://github.com/pydata/xarray/actions/workflows/upstream-dev-ci.yaml)
+    - [ ] Click on the most recent workflow and check that the `upstream-dev` job has run and passed. `upstream-dev` is not run on all all workflow runs.
```

Comment:
Would a failing upstream dev test block a release if it was not related to Zarr?

Reply:
[`c2f81db` (#3210)](https://github.com/zarr-developers/zarr-python/pull/3210/commits/c2f81dbfa7302c69a7193b87c639fa4df4058a41) updates the downstream component of the checklist to follow the structure that's worked pretty well for GMT. The developers of downstream libraries are pinged as part of the release checklist issue. Zarr-Python devs can decide whether to wait for all those downstream checks to be completed and whether any failures should be blocking.